spidermon/contrib/scrapy/monitors.py

```diff
@@ -27,6 +37,55 @@ def monitor_description(self):
         return super().monitor_description
 
 
+class BaseStatMonitor(BaseScrapyMonitor):
+    def run(self, result):
+        has_threshold_config = any(
+            [hasattr(self, "threshold_setting"), hasattr(self, "get_threshold")]
+        )
+        if not has_threshold_config:
+            raise NotConfigured(
+                f"{self.__class__.__name__} should include a a `threshold_setting` attribute "
+                "to be configured in your project settings with the desired threshold "
+                "or a `get_threshold` method that returns the desired threshold."
+            )
+
+        if (
+            hasattr(self, "threshold_setting")
+            and self.threshold_setting not in self.crawler.settings.attributes
+        ):
+            raise NotConfigured(
+                f"Configure {self.threshold_setting} to your project"
+                f"settings to use {self.monitor_name}."
+            )
+
+        return super().run(result)
+
+    def _get_threshold_value(self):
+        if hasattr(self, "get_threshold"):
+            return self.get_threshold()
+        return self.crawler.settings.get(self.threshold_setting)
+
+    def test_stat_monitor(self):
+        assertions = {
```

Comment:
I think we could simplify this by making a dictionary mapping the typical expression as a string to the assert method like so:

```python
def test_stat_monitor(self):
    assertions = {
           ">": self.assertGreater,
            ">=": self.assertGreaterEqual,
            "<": self.assertLess,
            "<=": self.assertLessEqual,
            "==": self.assertEqual,
            "!=": self.assertNotEqual
}
```

That way as a user I can do something like this for testing if the threshold if greater than or equal:

```python
class TestStatMonitor(BaseStatMonitor):
        stat_name = "test_stat"
        threshold_setting = "TEST_STAT_THRESHOLD"
        assert_type = ">="
```

Which might be clearer as we are using the standard expression for the comparison rather than going via a custom enum.

Reply:
@further-reading I liked your idea. It is much more easier for the user to know and define the type of the assertion. I will update as you suggested.